### PR TITLE
core(fr): add settings to context

### PR DESCRIPTION
**Pull request by adamraine**
Opening as a draft because I'm not sold on the way I made settings into a dependency.

Some other options I considered:
- Add `settings` directly to `FRTransitionalContext`
- Inject `settings` manually in `collectArtifactDependencies`

I converted `FullPageScreenshot` as an example of how this would be used.

### CODE REVIEW

**Comment by brendankenny** [comment]
> Opening as a draft because I'm not sold on the way I made settings into a dependency.

I haven't been in any of these conversations, but at least on first read it does seem maybe too far for the sake of conceptual purity ("all artifacts come from gatherers"), but then having to special case its initialization in `config.js` means it's somewhat harder to follow its lifetime instead of easier due to conceptual unification with all the other artifacts.

Is it worth re-thinking how we use the settings artifact? I think [the original reason](https://github.com/GoogleChrome/lighthouse/pull/4910/files#diff-25f864ed35c0eabc2a989689169c63148359a78c8a7bf93c7f871109b4781989R447-R448) for making it an artifact was for clarity in audits. It helps clear up which settings an audit is talking about since gathering and auditing can use different settings ([we never got around to allowing much difference](https://github.com/GoogleChrome/lighthouse/blob/1f27da747648fe52dcc09389d4c2fb6229d2632e/lighthouse-core/runner.js#L265-L267), but we still could), and we may just not have had `settings` on the audit context back then.

Since during gathering theres only one set of `settings`, having them on the context seems fine to me, personally. Then we could re-think if the artifact finalization step should sneak in `settings` into the artifacts at the very end or if audits could get settings access another way.

**Comment by adamraine** [comment]
The more I look at this, the more I am inclined to agree. We are going to need to handle settings differently in any situation. Creating a pseudo-gatherer to house settings just so it can be used as a dependency is overly convoluted. Other base artifacts do make sense as normal gatherers, but `settings` does not.

I also don't like the idea of adding a `Settings` shell artifact to the config, when the actual settings are defined elsewhere in the config.

**Comment by patrickhulce** [comment]
I'm also completely fine with `settings` existing on the context in gathering phase. The only reason it wasn't part of the initial context was I was unsure how much of settings would remain as-is. 

The important thing with settings, as @brendankenny already noted, is that the dependency on the *auditing* side is reflected, as the settings at gather mode time vs. audit time will in fact make a difference in some contexts.

File: lighthouse-core/test/fraggle-rock/config/config-test.js [hunk 1]
```diff
@@ -63,8 +63,8 @@ describe('Fraggle Rock Config', () => {
   });
 
   it('should throw on invalid artifact definitions', () => {
-    const configJson = {artifacts: [{id: 'FullPageScreenshot', gatherer: 'full-page-screenshot'}]};
-    expect(() => initializeConfig(configJson, {gatherMode})).toThrow(/FullPageScreenshot gatherer/);
+    const configJson = {artifacts: [{id: 'ScriptElements', gatherer: 'script-elements'}]};
```
Comment (adamraine) [inline lighthouse-core/test/fraggle-rock/config/config-test.js:66]: We are running out of candidates for this hehe

Comment (patrickhulce) [reply lighthouse-core/test/fraggle-rock/config/config-test.js:66]: yeah we are 😎

**Review by patrickhulce — APPROVED** [review]
LGTM, thanks @adamraine !

**Pull request merged**